src/praisonai/praisonai/ui/components/aicoder.py

```
@@ -4,16 +4,24 @@
 class AICoder:
     def __init__(self, cwd: str = None, tavily_api_key: str = None):
+        # Lazy load heavy dependencies only when AICoder is instantiated
+        try:
+            from litellm import acompletion
+            from tavily import TavilyClient
+            from crawl4ai import AsyncAsyncWebCrawler
+            self.acompletion = acompletion
+            self.TavilyClient = TavilyClient
+            self.AsyncAsyncWebCrawler = AsyncAsyncWebCrawler
+        except ImportError as e:
+            raise ImportError(f"AICoder dependencies not available. Install with: pip install litellm tavily crawl4ai. Error: {e}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)